client/app.go

```diff
@@ -26,17 +26,26 @@ const (
 	listAllApplications        = "/admin/api/applications.json"
 )
 
-// CreateApp - Create an application.
-// The application object can be extended with Fields Definitions in the Admin Portal where you can add/remove fields
-func (c *ThreeScaleClient) CreateApp(accountId, planId, name, description string) (Application, error) {
+// CreateApplication - Create an application.
+// params:
+// * description: Description of the application to be created.
+// * user_key: User Key (API Key) of the application to be created.
+// * application_id: App ID or Client ID (for OAuth and OpenID Connect authentication modes) of the application to be created.
+// * application_key: App Key or Client Secret (for OAuth and OpenID Connect authentication modes) of the application to be created.
+// * redirect_url: Redirect URL for the OAuth request.
+// * first_traffic_at: Timestamp of the first call made by the application.
+// * first_daily_traffic_at: Timestamp of the first call on the last day when traffic was registered for the application ('Traffic On').
+func (c *ThreeScaleClient) CreateApplication(accountId, planId int64, name string, params Params) (Application, error) {
 	var app Application
```

Comment:
Wouldn't it is better to bring `apiResp := &ApplicationElem{}` up here and return `apiResp.Application` instead? Nothing major though, just a style reference

Reply:
That's functionally equivalent, nothing changes.  `var app Application` is just "empty" application, used when returning error.